libraries/chain/webassembly/eos-vm-oc/executor.cpp

```diff
@@ -147,20 +148,42 @@ executor::executor(const code_cache_base& cc) {
    mapping_is_executable = true;
 }
 
-void executor::execute(const code_descriptor& code, const memory& mem, apply_context& context) {
+void executor::execute(const code_descriptor& code, memory& mem, apply_context& context) {
    if(mapping_is_executable == false) {
       mprotect(code_mapping, code_mapping_size, PROT_EXEC|PROT_READ);
       mapping_is_executable = true;
    }
 
+   uint64_t max_call_depth = eosio::chain::wasm_constraints::maximum_call_depth+1;
+   uint64_t max_pages = eosio::chain::wasm_constraints::maximum_linear_memory/eosio::chain::wasm_constraints::wasm_page_size;
+   if(context.control.is_builtin_activated(builtin_protocol_feature_t::configurable_wasm_limits)) {
+      const wasm_config& config = context.control.get_global_properties().wasm_configuration;
+      max_call_depth = config.max_call_depth;
+      max_pages = config.max_pages;
+   }
+   mem.reset(max_pages);
```

Comment:
Far too dangerous IMO. Could fail to acquire the resources. Also I think if the limit is large (like 2GB or something) it won't even work as-is.
Probably beyond a certain threshold (could be 33MB, could be more, could be less) OC should switch to using `mprotect()`.

Reply:
I changed it.  The threshold for switching to mprotect can be configured at construction.